src/EditorFeatures/Core/LanguageServer/AlwaysActiveLanguageClientEventListener.cs

```diff
@@ -42,6 +42,19 @@ public void StartListening(Workspace workspace)
     {
         // Trigger a fire and forget request to the VS LSP client to load our ILanguageClient.
         _ = LoadAsync();
+
+        _ = workspace.RegisterWorkspaceChangedHandler(Workspace_WorkspaceChanged);
+    }
+
+    private void Workspace_WorkspaceChanged(WorkspaceChangeEventArgs e)
+    {
+        if (e.Kind == WorkspaceChangeKind.SolutionAdded)
+        {
+            // We loaded the language client when the workspace started, above, but VS shuts it down when a solution closes,
+            // so when a new solution is loaded we have to load it again. The language server broker is smart enough not to
+            // load the same client multiple times, so reliability of this event is not a big concern.
+            _ = LoadAsync();
```

Comment:
aside.  @dibarbet these sorts of fire-and-forget changes, that don't use IAsyncOperationListener are likely bugs for future integration tests.

Reply:
loadasync has an async listener (and NFW)

Reply:
> loadasync has an async listener (and NFW)

itneresting.  that's technically not hte pattern we'd expect.  in that acse, *it* should not be async.

Reply:
You know if you could just add an event subscription to your PR, I wouldn't have to deal with a conflict 😛